.github/workflows/libabigail.yaml

```diff
@@ -0,0 +1,95 @@
+name: Libabigail ABI Checks
+on: 
+  pull_request: []
+
+jobs:
+  get-release:
+    container: ghcr.io/dyninst/dyninst-ubuntu-20.04:v12.1.0
+    runs-on: ubuntu-latest
+    steps:
+    - name: Upload Libs
+      uses: actions/upload-artifact@v2-preview
+      with:
+        name: release-libs
+        path: /opt/dyninst-env/.spack-env/view/lib
+
+  get-latest:
+    container: ghcr.io/dyninst/dyninst-ubuntu-20.04:latest
+    runs-on: ubuntu-latest
+    steps:
+    - name: Upload Libs
+      uses: actions/upload-artifact@v2-preview
+      with:
+        name: latest-libs
+        path: /opt/dyninst-env/.spack-env/view/lib
+
+  get-pr:
+    container: ghcr.io/dyninst/dyninst-ubuntu-20.04:latest
+    runs-on: ubuntu-latest
+    steps:
+    - name: Build Pull Request
+      uses: actions/checkout@v3
+    - name: Build
+      run: |
+         rm -rf /code
+         cp -R $PWD /code
+         ls /code
+         cd /opt/dyninst-env
+         /bin/bash build.sh         
+         
+    - name: Upload results
+      uses: actions/upload-artifact@v2-preview
+      with:
+        name: pr-libs
+        path: /opt/dyninst-env/.spack-env/view/lib
```

Comment:
3. yes! Just debugging. The reason is because if some version generates some libraries with unexpected names and the next step fails, you'd need this. it's good to preemptively include it, although for dyninst (so far) it's looked fairly consistent. 
4. technically for the single step, there isn't any loop over matrix.libs. Matrix.libs for one run (an entry in the side bar with a set of comparisons and one library name) is just for _one_ library name. The other libs are running in parallel (up to the max number of jobs allowed).

> and we're using the built-in failure detection for jobs instead of writing out own error-detction.

yep if libabigail has a non zero code you will see red.

> abi.strategy.fail-fast is 'false', then we'll always check all the libraries which may show that ABIs were broken in multiple places.

yep fail-fast false just says "if one job poops don't cancel the rest."

Yes, it is SO amazing I lost my mine with :exploding_head: when I wrote it and it like, actually worked! :laughing:

Reply:
ha maybe that's why they all passed.

Reply:
But are you sure? For spack environments they usually save the view to .spack-env - why would it be in install instead?

Reply:
>technically for the single step, there isn't any loop over matrix.libs. Matrix.libs for one run (an entry in the side bar with a set of comparisons and one library name) is just for one library name. The other libs are running in parallel (up to the max number of jobs allowed).

Aaah, ok. I thought about that a bit after I commented as the outermost property is 'job'. Does Github determine the max number of jobs allowed?

> But are you sure? For spack environments they usually save the view to .spack-env - why would it be in install instead?

build.sh only uses spack to provide access to the dependencies (boost, tbb, etc.). All of the build/install steps are done directly through CMake (https://github.com/dyninst/dyninst/blob/master/docker/build.sh#L20). I have more in-depth thoughts on this in https://github.com/dyninst/dyninst/pull/1222#issuecomment-1066034258.

Reply:
ah okay, if you want to do some suggests to the PR to update the paths we can see the new results! (and see if the ABI compat actually is failing).

Reply:
The only thing we need to change is line 44 from

    path: /opt/dyninst-env/.spack-env/view/lib
            
to

    path: /opt/dyninst-env/install/dyninst/lib

Everything else, we can figure out later.

Reply:
Why not latest too?

Also do you know how to suggest? There is an icon (first in the group on the right of one of these panels) where you can click and then it will let you type in to suggest changes for a line. That can be helpful in a PR when you know exactly what to change.

Reply:
Example

```suggestion
        path: /opt/dyninst-env/install/dyninst/lib
```
looks like:

![image](https://user-images.githubusercontent.com/814322/158082115-9ec30b72-ae85-467d-b2d2-f602883c80ac.png)

Reply:
>Why not latest too?

'latest' is built from the base container, so it has both the spack and build.sh installs. 'release' is a copy of 'latest', so that's good, too. This is super ugly and should definitely be homogenized, but we don't need to do that here. I'll try to work on the items I outlined in 1222 to see what I can do.

>Also do you know how to suggest?

I did not. That's super useful!

Reply:
>     'latest' is built from the base container, so it has both the spack and build.sh installs

Yep but if I'm following your logic, the build.sh is the one that actually has updated code?